src/Compilers/Core/Portable/WellKnownTypes.cs

```diff
@@ -337,6 +337,23 @@ internal enum WellKnownType
 
         System_Runtime_CompilerServices_ParamCollectionAttribute,
 
+        System_Linq_Expressions_BinaryExpression,
```

Comment:
Because we use well-known types now, there might be some behavior changes, right? For example, having ambiguous `BinaryExpression`s (so we won't resolve the well-known type for it), we now won't be able to bind `Expression.Binary` whereas previously we could. I'm not sure it matters though.

Reply:
> Because we use well-known types now, there might be some behavior changes, right? For example, having ambiguous `BinaryExpression`s (so we won't resolve the well-known type for it), we now won't be able to bind `Expression.Binary` whereas previously we could.

That is correct.

Reply:
This only happens when customers define a their own types in the `System` namespace though. That is indeed a break but it's generally one that we are okay with. Defining types in the `System` namespace, particularly when they clash with the ones in corelib is not recommended for this very reason